operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/ReconcileResult.java

```diff
@@ -2,26 +2,44 @@
 
 import java.util.Optional;
 
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+
 public class ReconcileResult<R> {
 
   private final R resource;
   private final Operation operation;
+  private final Exception error;
```

Comment:
I don't think we should have this errors, we should just trow it and initiate a retry.

Reply:
That's an option yes but the idea here is so that a dependent that's failing doesn't prevent the other ones from doing their work. The main reconciler could then decide what to do with the dependent. Without this change, if a managed dependent is failing your main reconciler might never get called. It should be up to the reconciler to decide what to do with the failing dependent.

Reply:
But that would mean that the reconciler should always check for error for each dependent.
This does not fit with the current error handling approach either, so currently nobody does a `try-cach` on every fabric8 client action. Also some might just simply don't do the other reconciliation if the one before failed.

Reply:
We could have it at the controller level: check if any of the dependents is in error and throw a composite error if some are.

Reply:
Also, it could be argued that it's a good practice for reconcilers to check the status of each dependent, which is the whole point of `ReconcileResult`.

Reply:
But this contradicts with whole current concept of `ErrorStatusHanlder`, that we have a one cross cutting place to handle errors. It simply does not fit in the picture. (Not speaking that the samples are not migrated to this in the PR). 
So also goes agains the concept that dependent resources seamlesly fit into the current concepts (well in this case the current error handling approach.

(And how to force this for standalone dependent resources? )

So I would not introduce such conceptual change, with so many open questions before release. We can discuss this further in depth later, how to handle this.

Reply:
> Also, it could be argued that it's a good practice for reconcilers to check the status of each dependent, which is the whole point of ReconcileResult.

It was not the point until now :) since there was not even an API, you could check if it was update created or none, but error handling is completely different story.